components/match2/wikis/fortnite/match_group_input_custom.lua

```diff
@@ -21,31 +22,100 @@ local MatchFunctions = {
 	},
 	DEFAULT_MODE = 'team'
 }
+local FfaMatchFunctions = {}
+local MapFunctions = {}
+local FfaMapFunctions = {}
 
-local CustomMatchGroupInput = {}
+local DEFAULT_BESTOF = 3
 
 ---@param match table
 ---@param options table?
 ---@return table
 function CustomMatchGroupInput.processMatch(match, options)
-	return MatchGroupInputUtil.standardProcessFfaMatch(match, MatchFunctions)
+	return MatchGroupInputUtil.standardProcessMatch(match, MatchFunctions, FfaMatchFunctions)
+end
+
+--- Normal 2-opponent Match
+
+---@param match table
+---@param opponents table[]
+---@return table[]
+function MatchFunctions.extractMaps(match, opponents)
+	return MatchGroupInputUtil.standardProcessMaps(match, opponents, MapFunctions)
+end
+
+---@param match table
+---@param games table[]
+---@param opponents table[]
+---@return table
+function MatchFunctions.getExtraData(match, games, opponents)
+	return {
+		mvp = MatchGroupInputUtil.readMvp(match, opponents),
+		casters = MatchGroupInputUtil.readCasters(match),
+	}
+end
+
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer?
+function MatchFunctions.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
+	end
+end
+
+---@param bestofInput string|integer?
+---@return integer
+function MatchFunctions.getBestOf(bestofInput)
+	local bestof = tonumber(bestofInput) or tonumber(Variables.varDefault('match_bestof')) or DEFAULT_BESTOF
+	Variables.varDefine('match_bestof', bestof)
+	return bestof
+end
+
+---@param games table[]
+---@return table[]
+function MatchFunctions.removeUnsetMaps(games)
+	return Array.filter(games, function(map)
```

Comment:
why check for map.map being set?
can it happen that it is unset in a valid map?
maybe an Logic.isNotDeepEmpty check would be better

Reply:
I've removed this one because not sure what it does from the TFT setup, but removing seems to not causing error

Reply:
it is needed so you do not store empty maps data in lpdb
my question was why it specifically checked for a set map instead of checking for non empty input, i.e. something like:
```lua
---@param games table[]
---@return table[]
function MatchFunctions.removeUnsetMaps(games)
	return Array.filter(games, Logic.isNotDeepEmpty)
end
```